cmd/internal/find/primary/meta/objectPredicate.go

```diff
@@ -0,0 +1,79 @@
+package meta
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/puppetlabs/wash/cmd/internal/find/primary/errz"
+)
+
+// A key consists of one or more characters that aren't
+// ".", "[", or "]". The reason we have this limitation
+// is so we can support key sequences, which includes
+// nested keys like ".key1.key2", ".key1[].key2", etc.
+// key sequences aren't specified in the grammar because
+// they make it harder to formalize the semantics.
+var keyRegex = regexp.MustCompile(`([^\.\[\]]+)`)
+
+// ObjectPredicate => EmptyPredicate | ‘.’ Key Predicate
+// Key             => keyRegex
+func parseObjectPredicate(tokens []string) (predicate, []string, error) {
+	if p, tokens, err := parseEmptyPredicate(tokens); err == nil {
+		return p, tokens, err
+	}
+	// EmptyPredicate did not match, so try '.' Key Predicate
+	if len(tokens) == 0 {
+		return nil, nil, errz.NewMatchError("expected a key sequence")
+	}
+	tk := tokens[0]
+	if tk[0] != '.' {
+		return nil, nil, errz.NewMatchError("key sequences must begin with a '.'")
+	}
+	tk = tk[1:]
+	loc := keyRegex.FindStringIndex(tk)
```

Comment:
This seems to allow a key sequence that doesn't immediately follow the `.`, because the regex doesn't start with `^`.

Reply:
What would a valid instance of this look like?

Reply:
Hmm, was thinking of being as flexible as possible there, but the regex does allow for spaces and such (on top of missing the `^`). I'll revise this later.

Given `.key1`, valid instance would be `key1`. Given `.key1.key2`, valid instance would be `key1`; postmatch would be `.key2`